test/fail_compilation/fail17969.d

```diff
@@ -1,6 +1,6 @@
 /* TEST_OUTPUT:
 ---
-fail_compilation/fail17969.d(9): Error: no property `sum` for type `MapResult2!((b) => b)`
+fail_compilation/fail17969.d(9): Error: no property `sum` for type `fail17969.__lambda6!(int[]).__lambda6.MapResult2!((b) => b)`
```

Comment:
Perhaps we should special case Voldemort types after all.

Reply:
BTW, the type of errors that one typically gets when missing std.algorithm / std.range imports are surprisingly underrepresented in the dmd test suite.

Reply:
I'm not sure we should. At least with this change you'd know where to look to find the specific Voldermort type.

Reply:
If you do that, in this particular situation, if you instantiate with another type (other then `int`) you would have the same error message at the same line, giving you no hint at all. With this patch you will have additional information(with what types the template has been instantiated). Ideally, the instantiation location should also be outputted, but I think that this should be the topic of a different PR.

Reply:
To be honest, I didn't check the test case in question. I assumed that it follows a typical phobos Voldemort pattern, but there are no Voldemort types involved at all. Disregard my comment.